pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -185,7 +186,7 @@ func (t ResolvedPipelineTask) isFailure() bool {
 		atLeastOneFailed := false
 		for _, taskRun := range t.TaskRuns {
 			isDone = isDone && taskRun.IsDone()
-			taskRunFailed := taskRun.Status.GetCondition(apis.ConditionSucceeded).IsFalse() && !t.hasRemainingRetries()
+			taskRunFailed := taskRun.Status.GetCondition(apis.ConditionSucceeded).IsFalse() && !t.isCustomTaskRetriable()
```

Comment:
this is for case is handling taskruns only - so it shouldn't check `isCustomTaskRetriable` 

matrixed custom tasks are handled in the first case of the switch statement

Reply:
SG. updated!